resources/Microsoft.Windows.Setting.Language/Microsoft.Windows.Setting.Language.psm1

```diff
@@ -0,0 +1,213 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+using namespace System.Collections.Generic
+
+if ([string]::IsNullOrEmpty($env:TestRegistryPath)) {
+    $global:LocaleNameRegistryPath = 'HKCU:\Control Panel\International'
+    $global:LocaleUserProfilePath = 'HKCU:\Control Panel\International\User Profile'
+} else {
+    $global:LocaleNameRegistryPath = $global:LocaleUserProfilePath = $env:TestRegistryPath
+}
+
+#region Functions
+function Get-OsBuildVersion {
+    return [System.Environment]::OSVersion.Version.Build
+}
+
+function TryGetRegistryValue {
+    param (
+        [Parameter(Mandatory = $true)]
+        [string]$Key,
+
+        [Parameter(Mandatory = $true)]
+        [string]$Property
+    )    
+
+    if (Test-Path -Path $Key) {
+        try {
+            return (Get-ItemProperty -Path $Key | Select-Object -ExpandProperty $Property)     
+        } catch {
+            Write-Verbose "Property `"$($Property)`" could not be found."
+        }
+    } else {
+        Write-Verbose 'Registry key does not exist.'
+    }
+}
+
+function Get-LocaleList {
+    $localeList = Get-WinUserLanguageList
+    $out = [List[Language]]::new()
+
+    foreach ($locale in $localeList) {
+        $language = [Language]::new($locale.LanguageTag, $true)
+        $out.Add($language)
+    }
+
+    # section to include other languages that can be installed
+    # helpful for users to discover what packages can be installed
+    $allLangues = [System.Globalization.CultureInfo]::GetCultures('AllCultures')
+    foreach ($culture in $allLangues) {
+        if ($out.LocaleName -notcontains $culture.Name -and -not ([string]::IsNullOrEmpty($culture.Name))) {
```

Comment:
Does $out.LocaleName return a collection of localeNames? I feel that this will be an error. Also whether $culture.Name is null/empty should be checked first in this if statement.